### PR TITLE
Prototype for Redis exception injection

### LINKED ISSUE #102: Working prototype for exception injection
Using lettuce interposition, inject exceptions using Filibuster that throw errors.

This should cause your test to fail, if your test fails to handle these errors when the cache is primed.

### PR DESCRIPTION
**Pull request by MichaelA-ops**
closes #102

### CODE REVIEW

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisAccessTest.java [hunk 1]
```diff
@@ -62,27 +70,75 @@ public void testRedisHello() {
...
-    @DisplayName("Tests whether Redis interceptor can inject a timeout exception")
+    @DisplayName("Tests whether Redis sync interceptor can inject a timeout exception")
     @Order(2)
-    public void testRedisInterceptedTimeoutExceptionThrowing() {
-        RedisCommands<Object, Object> myRedisCommands = LettuceInterceptedConnection.create(redisConnection);
-        LettuceInterceptor.isFaultInjected = true;
-        String key = "test";
-        String value = "example";
+    public void testRedisSyncException() {
+        RedisCommands<String, String> myRedisCommands = getRedisConnection(RedisCommands.class, true);
```
Comment (cmeiklejohn) [inline src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisAccessTest.java:91]: this is good, I like how the type is the same when we are injecting a fault

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisAccessTest.java [hunk 2]
```diff
@@ -62,27 +70,75 @@ public void testRedisHello() {
         }
     }
 
+    private <T> T getRedisConnection(Class<T> type, boolean isFaultInjected) {
```
Comment (cmeiklejohn) [inline src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisAccessTest.java:73]: anyway the T in Class<T> can be more specific?  surely it has to implement some interface, right?

Comment (cmeiklejohn) [reply src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisAccessTest.java:73]: if not, that's fine

**Review by cmeiklejohn — APPROVED** [review]

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisAccessTest.java (hunk 2 above)
Comment (MichaelA-ops) [reply src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisAccessTest.java:73]: RedisCommands, RedisAsyncCommands and RedisReactiveCommands don't share any interfaces

**Pull request merged**